### ISSUE TITLE
putobject functional test fails for s3 gateway

### ISSUE THREAD

**Issue by poornas** [minio/minio-py#578]
Mint run against minio release build of Sep 25 fails for  s3 gateway
```
Traceback (most recent call last):
  File "/mint/run/core/minio-py/tests.py", line 333, in <module>
    main()
  File "/mint/run/core/minio-py/tests.py", line 257, in main
    object_name).get_exception()
minio.error.InternalError: InternalError: message: We encountered an internal error. Please try again.
```

Mint image built from  commit 158108152fc86c52d5543995e0e7634bf1b6da74
Minio version:
Version: 2017-09-25T21:57:17Z
Release-Tag: RELEASE.2017-09-25T21-57-17Z
Commit-ID: d3eb5815d9951fdb82d0134b7651368bab71d72e

**Comment by poornas** [comment]
There was a regression on minio server that causes this failure in gateway mode - PR https://github.com/minio/minio/pull/4972 fixes this on minio server . Tested against the PR's code base and functional tests run okay.

**Comment by vadmeste** [comment]
@poornas, are you sure that https://github.com/minio/minio/pull/4972 fixes the issue ? That PR is closed by the way. 

I tested with last master but the error is there. I think Minio should vendorize new change in minio-go first.

**Comment by harshavardhana** [comment]
> I tested with last master but the error is there. I think Minio should vendorize new change in minio-go first.

We need this https://github.com/minio/minio/pull/4989

**Comment by poornas** [comment]
@vadmeste , thanks for the heads up - didn't notice that minio/minio#4972 got closed.Will investigate again..

**Comment by poornas** [comment]
With minio/minio#4989, tests pass. failure was due to invalid content-md5

**Comment by vadmeste** [comment]
> With minio/minio#4989, tests pass. failure was due to invalid content-md5

yes, I confirm.

**Comment by poornas** [comment]
@deekoder , this can be closed as the issue is fixed by  minio/minio#4989 which has been merged.
